q2_fmt/_peds.py

```diff
@@ -468,3 +484,89 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
 def _mask_recipient(donor_mask, recip_df):
     maskedrecip = donor_mask & recip_df
     return maskedrecip
+
+
+def peds_bootstrap(table: pd.DataFrame, metadata: qiime2.Metadata,
+                   time_column: str, reference_column: str,
+                   subject_column: str,
+                   filter_missing_references: bool = False,
+                   drop_incomplete_subjects: bool = False,
+                   drop_incomplete_timepoint: list = None,
+                   bootstrap_replicates: int = 999) -> (pd.DataFrame):
+
+    metadata_df = metadata.to_dataframe()
+    donor = metadata_df[metadata_df.index.isin(
+        _check_reference_column(metadata=metadata_df,
+                                reference_column=reference_column))]
+
+    recipient = metadata_df.loc[metadata_df[reference_column].notnull()]
+    fake_donor = pd.DataFrame([])
+    for i in range(0, bootstrap_replicates+1):
+        if i == 0:
+            peds = sample_peds(
+                table=table, metadata=metadata,
+                time_column=time_column,
+                reference_column=reference_column,
+                subject_column=subject_column,
+                filter_missing_references=filter_missing_references,
+                drop_incomplete_subjects=drop_incomplete_subjects,
+                drop_incomplete_timepoint=drop_incomplete_timepoint)
+            real_temp = peds["measure"]
+        else:
+            shifted_list = recipient[reference_column].sample(frac=1).to_list()
```

Comment:
I recommend calling this "shuffled" list, as "shifted" would imply something different. The operation we're carrying out here is like shuffling a deck of cards, not (for example) shifting every item's index in a list by adding one to it.

Reply:
And would it work to bypass creating this new list all together? For example, [this](https://stackoverflow.com/a/34879805) SO post shows how to do this operation in place. It looks like you don't use the metadata after this, so you could keep shuffling it over and over.

Reply:
Re: not creating the list. Those examples dont quite do what I am doing because those shuffled lists keep the ID and value tied together so they are more just shuffling the rows in the dataframe. 

The list allows me to keep the sample ids(index) the same and shuffle the refernece list.  I will look in to a simpler way to do this.